src/borg/_hashindex.c

```diff
@@ -111,7 +111,7 @@ hashindex_index(HashIndex *index, const void *key)
 static int
 hashindex_lookup(HashIndex *index, const void *key)
```

Comment:
> we could bail after at most max_offset iterations.
Actually if the offset of the key is smaller than the number of buckets we've looked at we can bail.  There's no way the next bucket will contain out desired key, since it would have been swapped on insert.